python/lsst/pipe/tasks/setPrimaryFlags.py

```diff
@@ -36,25 +36,31 @@ class SetPrimaryFlagsConfig(Config):
 class SetPrimaryFlagsTask(Task):
     ConfigClass = SetPrimaryFlagsConfig
 
-    def __init__(self, schema, **kwargs):
+    def __init__(self, schema, isSingleFrame=False, **kwargs):
         Task.__init__(self, **kwargs)
         self.schema = schema
-        self.isPatchInnerKey = self.schema.addField(
-            "detect_isPatchInner", type="Flag",
-            doc="true if source is in the inner region of a coadd patch",
-        )
-        self.isTractInnerKey = self.schema.addField(
-            "detect_isTractInner", type="Flag",
-            doc="true if source is in the inner region of a coadd tract",
-        )
+        self.isSingleFrame = isSingleFrame
+        if not self.isSingleFrame:
+            self.isPatchInnerKey = self.schema.addField(
+                "detect_isPatchInner", type="Flag",
+                doc="true if source is in the inner region of a coadd patch",
+            )
+            self.isTractInnerKey = self.schema.addField(
+                "detect_isTractInner", type="Flag",
+                doc="true if source is in the inner region of a coadd tract",
+            )
+            primaryDoc = ("true if source has no children and is in the inner region of a coadd patch "
+                          "and is in the inner region of a coadd tract "
+                          "and is not \"detected\" in a pseudo-filter (see config.pseudoFilterList)")
+        else:
+            primaryDoc = "true if source has no children and is not a sky source"
         self.isPrimaryKey = self.schema.addField(
             "detect_isPrimary", type="Flag",
-            doc="true if source has no children and is in the inner region of a coadd patch "
-                "and is in the inner region of a coadd tract "
-                "and is not \"detected\" in a pseudo-filter (see config.pseudoFilterList)",
+            doc=primaryDoc,
```

Comment:
Do you need both "doc" and "primaryDoc"? Where do these each appear? I also noticed the coadd primary doc does not specify anything about sky sources, while the single frame one does. Is that intentional?

Reply:
In this case, the doc string for `SetPrimaryFlagsTask` is different depending on whether or not `self.isSingleFrame` is `True` (line 55) or `False` (line 43). If true, the doc string refers only to children and sky sources (or pseudo-filters, see below), however, if false, the doc string additionally refers to coadd patch and tract selections (which would not make sense in the single frame output). We use the `if` logic above to define `primaryDoc`, which is then subsequently placed into the main `self.isPrimaryKey` doc string. I think that's the best way to do it, but happy to consider another approach if you have something in mind? 

With regards not mentioning sky sources in the coadd primary doc, the coadd primary doc refers to "pseduo-filters", of which sky sources are included. I agree that the language here may not be immediately obvious, but this was a decision made prior to this ticket and so we were reluctant to change it. I think the logic is that other additional 'pseudo-filters' can be added in addition to sky sources which would then be picked up by this task at the coadd level, but currently, only sky sources are implemented.

Reply:
Ah, that is subtle! The order in which they're defined threw me off. For coadds, you define primaryDoc after the two docs because it includes them both. Can you follow this order for single frame even though it's not necessary, for consistency?

Reply:
Sorry, I don't quite follow. The two extra docs being defined for coadds are `isPatchInnerKey` and `isTractInnerKey`. These docs don't apply in the single frame case, so they're not being set there (so there's nothing to move `primaryDoc` above or below of). The single frame `__init__` doc string is just the `primaryDoc`, which then gets added to `isPrimaryKey`. Sorry if I've missed something here, please let me know!

Reply:
Sorry :) I'm suggesting delete line 56, explicitly set `doc = "true if source has no children and is not a sky source"` on line 59, and then explicitly define `primaryDoc` as the last thing.

Reply:
Thanks again for the feedback here. Marking the issue as resolved following our earlier discussion, cheers!